pandas/io/excel.py

```diff
@@ -200,8 +200,12 @@ def read_excel(io, sheetname=0, header=0, skiprows=None, skip_footer=0,
     if not isinstance(io, ExcelFile):
         io = ExcelFile(io, engine=engine)
 
+    # maintain backwards compatibility by converting sheetname to sheet_name
+    if 'sheetname' in kwds:
+         sheet_name = kwds.pop('sheetname')
```

Comment:
needs a deprecation warning on the original arg. You need to *also* accept the original arg as a kwarg.